### PR TITLE
Improve error message when CanvasKit is unable to parse a font

### PR DESCRIPTION
**Pull request by hterkelsen**
Fixes https://github.com/flutter/flutter/issues/76804

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: lib/web_ui/lib/src/engine/canvaskit/fonts.dart [hunk 1]
```diff
@@ -141,7 +148,15 @@ class SkiaFontCollection {
...
 
     final Uint8List bytes = buffer.asUint8List();
-    return _RegisteredFont(bytes, family);
+    SkTypeface? typeface =
+        canvasKit.FontMgr.RefDefault().MakeTypefaceFromData(bytes);
+    if (typeface != null) {
+      return _RegisteredFont(bytes, family, typeface);
+    } else {
+      html.window.console.warn('Failed to load font $family at $url');
+      html.window.console.warn('Verify that $url contains a valid font.');
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/fonts.dart:157]: Let's add an indirection, e.g. a mutable top-level `void Function() printWarning;` that could be mocked in a test so we can test that we log these warnings.

**Review by yjbanov — COMMENTED** [review]
The code LGTM, but this PR needs a test.

File: lib/web_ui/lib/src/engine/canvaskit/fonts.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/canvaskit/fonts.dart]: Done. PTAL

**Review by yjbanov — APPROVED** [review]
lgtm

**Review by yjbanov — APPROVED** [review]

**Pull request merged**